gcloud/bigtable/client.py

```diff
@@ -0,0 +1,112 @@
+
+"""Parent client for calling the Google Cloud Bigtable API.
+
+This is the base from which all interactions with the API occur.
+
+In the hierarchy of API concepts
+
+* a :class:`Client` owns a :class:`.Cluster`
+* a :class:`.Cluster` owns a :class:`Table <gcloud_bigtable.table.Table>`
+* a :class:`Table <gcloud_bigtable.table.Table>` owns a
+  :class:`ColumnFamily <.column_family.ColumnFamily>`
+* a :class:`Table <gcloud_bigtable.table.Table>` owns a :class:`Row <.row.Row>`
+  (and all the cells in the row)
+"""
+
+
+from gcloud.client import JSONClient
+
+
+TABLE_ADMIN_HOST = 'bigtabletableadmin.googleapis.com'
+"""Table Admin API request host."""
+TABLE_ADMIN_PORT = 443
+"""Table Admin API request port."""
+
+CLUSTER_ADMIN_HOST = 'bigtableclusteradmin.googleapis.com'
+"""Cluster Admin API request host."""
+CLUSTER_ADMIN_PORT = 443
+"""Cluster Admin API request port."""
+
+DATA_API_HOST = 'bigtable.googleapis.com'
+"""Data API request host."""
+DATA_API_PORT = 443
+"""Data API request port."""
+
+ADMIN_SCOPE = 'https://www.googleapis.com/auth/cloud-bigtable.admin'
+"""Scope for interacting with the Cluster Admin and Table Admin APIs."""
+DATA_SCOPE = 'https://www.googleapis.com/auth/cloud-bigtable.data'
+"""Scope for reading and writing table data."""
+READ_ONLY_SCOPE = ('https://www.googleapis.com/auth/'
+                   'cloud-bigtable.data.readonly')
+"""Scope for reading table data."""
+
+
+class Client(JSONClient):
+    """Client for interacting with Google Cloud Bigtable API.
+
+    .. note::
+
+        Since the Cloud Bigtable API requires the gRPC transport, no
+        ``http`` argument is accepted by this class.
+
+    :type project: :class:`str` or :func:`unicode <unicode>`
+    :param project: (Optional) The ID of the project which owns the
+                    clusters, tables and data. If not provided, will
+                    attempt to determine from the environment.
+
+    :type credentials:
+        :class:`OAuth2Credentials <oauth2client.client.OAuth2Credentials>` or
+        :data:`NoneType <types.NoneType>`
+    :param credentials: (Optional) The OAuth2 Credentials to use for this
+                        cluster. If not provided, defaulst to the Google
+                        Application Default Credentials.
+
+    :type read_only: bool
+    :param read_only: (Optional) Boolean indicating if the data scope should be
+                      for reading only (or for writing as well). Defaults to
+                      :data:`False`.
+
+    :type admin: bool
+    :param admin: (Optional) Boolean indicating if the client will be used to
+                  interact with the Cluster Admin or Table Admin APIs. This
+                  requires the :const:`ADMIN_SCOPE`. Defaults to :data:`False`.
+
+    :raises: :class:`ValueError <exceptions.ValueError>` if both ``read_only``
+             and ``admin`` are :data:`True`
+    """
+
+    # Prevent creation of Client.connection.
+    _connection_class = None
```

Comment:
Hmm, how does that work?  `Client.__init__()` wants to call `self._connection_class(credentials, http)`.